scte35/segmentationdescriptor.go

```diff
@@ -255,19 +257,20 @@ func (d *segmentationDescriptor) CanClose(out SegmentationDescriptor) bool {
 			return true
 		}
 	case segCloseEventIDNotNested:
+		// this should also consider segnum == segexpected for IN signals closing an out signal.
+		if d.IsIn() && d.EventID() == out.EventID() && d.SegmentNumber() == d.SegmentsExpected() {
+			return true
+		}
 		if d.EventID() != out.EventID() {
 			return false
 		}
 		fallthrough
```

Comment:
Is this fallthrough still needed?  From cog esam 2.0 doc, it seems like the logic has diverged enough that only the first if block is needed and all the old code can be deleted.

Reply:
The `if d.EventId()...` statement is extraneous.  As is the `fallthrough`.